src/fheroes2/army/army_ui_helper.cpp

```diff
@@ -88,12 +120,26 @@ void fheroes2::drawMiniMonsters( const Troops & troops, int32_t cx, const int32_
             fheroes2::Blit( monster, output, cx + offset, cy + offsetY + monster.y() );
             text.draw( cx + chunk - text.width() - offset, cy + 23, output );
         }
-        else {
+        else if ( slotsToSkip == 0 ) {
             const int offsetY = 28 - monster.height();
-            fheroes2::Blit( monster, output, cx - monster.width() / 2 + monster.x() + 2, cy + offsetY + monster.y() );
-            text.draw( cx - text.width() / 2, cy + 29, output );
+            int offsetX = -14;
+
+            // Center the monster if there is only one
+            if ( count == 1 && slot == 1 ) {
+                offsetX = -10;
+            }
+
+            int x = ( cx - ( monster.width() / 2 ) ) + offsetX;
+            int y = cy + offsetY + monster.y();
```

Comment:
`fheroes2::Blit()` accepts `int32_t` so to avoid unnecessary type conversions lets set these variables type to `int32_t`.
```suggestion
            int32_t x = ( cx - ( monster.width() / 2 ) ) + offsetX;
            int32_t y = cy + offsetY + monster.y();
```